### PR TITLE
Added support for inter-glyph space.

### LINKED ISSUE #868: Bug in element determination
I found a major and long-standing bug in the code.  In `gabc-elements-determination.c`, there is this code:

```
        if (is_puncta_ascendens(current_glyph->type)) {
            current_glyph_type = G_PUNCTA_ASCENDENS;
        } else {
            if (is_puncta_descendens(current_glyph->type)) {
                current_glyph_type = G_PUNCTA_DESCENDENS;
            } else {
                current_glyph_type = current_glyph->type;
            }
        }
```

The problem is, `current_glyph_type` is being tested against `gregorio_glyph_type` values but `current_glyph->type` is of type `gregorio_type`, and in this case is *always* `GRE_GLYPH` (which has the same numeric value as `G_2_PUNCTA_INCLINATA_DESCENDENS`).

So this code is *unconditionally* setting `current_glyph_type` to `G_PUNCTA_DESCENDENS` before making any further element determination.

If I fix this by changing the code to
```
        if (is_puncta_ascendens(current_glyph->u.notes.glyph_type)) {
            current_glyph_type = G_PUNCTA_ASCENDENS;
        } else {
            if (is_puncta_descendens(current_glyph->u.notes.glyph_type)) {
                current_glyph_type = G_PUNCTA_DESCENDENS;
            } else {
                current_glyph_type = current_glyph->u.notes.glyph_type;
            }
        }
```
... it changes **A LOT** of tests.  Do I fix it?

This code has literally been doing this since *inception* (see https://github.com/gregorio-project/gregorio/blob/588be00ed31072c8d0c2ef14d382718bdd130ecf/plugins/gabc/gabc-elements-determination.c).

### PR DESCRIPTION
**Pull request by henryso**
Fixes #868.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ As of v3.0.0 this project adheres to [Semantic Versioning](http://semver.org/).
...
 - It is now possible to generate a font with a different stem length schema (apart from the default one, Solesmes' is available), see [fonts/README.md](fonts/README.md) for details.
 - Some forms (`ed` for example) will have a different stem length if a ledger line is present. Heuristics to determine the presence of a ledger line or not are quite simple and may not always be correct. Gabc now allows you to force a note to act as if a ledger line was present with `[ll:1]`, or as if there was no ledger line, with `[ll:0]`. See Gabc section of GregorioRef for details.
+- The ability to typeset an inter-glyph space using `\!` in gabc (see [#868](https://github.com/gregorio-project/gregorio/issues/868))
```
Comment (eroux) [inline CHANGELOG.md]: I think you mean `/!` instead of `\!`?

**Comment by henryso** [comment]
Thanks for noticing.

**Comment by eroux** [comment]
I'm just wondering if it would be better to have `SP_INTERGLYPH_SPACE` explicitely in the structure when relevant, like in `(gg)`?... This would make gabc output use it every time... But I'm not really sure, what do you think?

**Comment by henryso** [comment]
I don't really see the benefit of adding it into the structure.

**Comment by eroux** [comment]
Ok, let's merge it then, thanks a lot!

**Pull request merged**